components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -486,88 +486,361 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
-                        userNames = getUserListOfRoles(roleNames);
-                    } else {
-                        throw new NotImplementedException(String.format("Filter operator %s is not supported " +
-                                "by the user store.", filterOperation));
-                    }
+                    // Extract he domain name if the domain name is embedded in the filter attribute value.
+                    domainName = resolveDomainNameInAttributeValue(domainName, node);
+
+                    // Get filtered user names.
+                    userNames = getFilteredUsers(node, offset, limit, sortBy, sortOrder, domainName);
                 }
-            } catch (UserStoreException e) {
-                throw new CharonException(String.format("Error in filtering users by attribute name: %s, " +
-                                "attribute value: %s and filter operation %s", attributeName, attributeValue,
-                        filterOperation), e);
+            } else
+                userNames = filterUsersUsingLegacyAPIs(node, limit, offset);
+        } catch (NotImplementedException e) {
+            String errorMessage = String.format("System does not support filter operator: %s", node.getOperation());
+            throw new CharonException(errorMessage, e);
+        } catch (BadRequestException e) {
+            String errorMessage = String
+                    .format("Domain parameter: %s in request does not match with the domain name in the attribute "
+                            + "value: %s ", domainName, node.getValue());
+            throw new CharonException(errorMessage, e);
+        } catch (CharonException e) {
+            String errorMessage = String.format("Error in filtering users by attribute name: %s, "
+                            + "attribute value: %s and filter operation: %s", node.getAttributeValue(), node.getValue(),
+                    node.getOperation());
+            throw new CharonException(errorMessage, e);
+        }
+        return getDetailedUsers(userNames, requiredAttributes);
+    }
+
+    /**
+     * Method to handle a request without the count(limit). Since the new APIs require a limit, all the users in a
+     * user store cannot be fetched. Therefore, legacy APIs have been used to retrieve all the users in a user store.
+     *
+     * @param node       Expression node
+     * @param limit      Results count
+     * @param offset     Starting index of the filter
+     * @param domainName Domain name in the request
+     * @return User list with their usernames
+     * @throws CharonException
+     * @throws NotImplementedException
+     */
+    private String[] getAllFilteredUsers(ExpressionNode node, int limit, int offset, String domainName)
+            throws CharonException, NotImplementedException {
+
+        // If there is a domain, append the domain with the domain separator in front of the new attribute value.
+        if (StringUtils.isNotEmpty(domainName))
```

Comment:
Cover the if block with `{}`. Do this for all the places.

Reply:
This is changed in the current master branch